core/src/main/scala/cats/mtl/instances/chronicle.scala

```diff
@@ -0,0 +1,68 @@
+package cats
+package mtl
+package instances
+
+import cats.data.{Ior, IorT}
+import cats.syntax.functor._
+
+trait ChronicleInstances extends ChronicleLowPriorityInstances {
+  implicit final def chronicleInd[M[_], Inner[_], E](implicit ml: MonadLayerControl[M, Inner],
+                                                     under: MonadChronicle[Inner, E]
+                                                    ): MonadChronicle[M, E] = {
+    new DefaultMonadChronicle[M, E] {
+      val monad: Monad[M] =
+        ml.outerInstance
+
+      def dictate(c: E): M[Unit] = {
+        ml.layer(under.dictate(c))
+      }
+
+      def confess[A](c: E): M[A] =
+        ml.layer(under.confess(c))
+
+      def materialize[A](fa: M[A]): M[Ior[E, A]] =
+        ml.outerInstance.flatMap(ml.layerControl { nt =>
+          under.materialize(nt(fa))
+        }) {x => x.traverse(ml.restore)(ml.outerInstance) }
```

Comment:
This method is really complex and does quite a lot and might therefore be quite inefficient. Maybe it's better to have a different primitive after all and derive `materialize` from there. Sorry for suggesting so many changes.
`def overcome[A](ma: M[A])(f: Either[E, (E, A)] => A): M[A]`. Then this would go away as lifting it would just be ```scala
ml.layerMapK(ma)((new (Inner ~> Inner) {
  def apply[X](fa: Inner[X]): Inner[X] = under.overcome(fa)(f)
})
```

That should also give us the ability to remove the need for `MonadLayerControl` :)

Reply:
Disregard this, `under.overcome` would need a function ` Either[E, (E, X)] => X`, so this won't work and we're stuck with this it seems :smile: